TransformationSystem/Agent/TaskManagerAgentBase.py

```diff
@@ -541,7 +553,34 @@ def submitTasks(self, transIDOPBody, clients):
     self._logInfo("Obtained %d tasks for submission" % len(tasks),
                   method=method, transID=transID)
 
-    # Prepare tasks
+    # Prepare tasks and submits them, by chunks
+    chunkSize = self.maxParametricJobs if self.bulkSubmissionFlag else self.tasksPerLoop
+    for taskDictChunk in breakDictionaryIntoChunks(tasks, chunkSize):
+      res = self._prepareAndSubmitAndUpdateTasks(transID, transBody, taskDictChunk,
+                                                 owner, ownerDN, ownerGroup,
+                                                 clients)
+      if not res['OK']:
+        return res
+      self._logVerbose("Submitted %d jobs, bulkSubmissionFlag = %s" % (chunkSize, self.bulkSubmissionFlag))
```

Comment:
`chunkSize` --> `len(taskDictChunk)` for the case when `tasks` does not hold a multiple of `chunkSize` and the last slice is smaller.

Out of curiosity, what is the value of tasksPerLoop and MaxParametricJobs you have in your configuration?

Reply:
Thanks, I will correct. 

We have tasksPerLoop=100 and MaxParametricJobs is not set in CS: the default is 20.